### PR TITLE
elector: fix elector goroutine leak

### PR DESCRIPTION
**Pull request by jdef**
See title. No APIs were harmed in the crafting of this PR.

https://jira.mesosphere.com/browse/DCOS-40480

### CODE REVIEW

**Comment by jdef** [comment]
oops, quickly fixed/squashed code formatting issue

**Comment by jdef** [comment]
.. and fix another naming thing to pacify the linter

File: elector/elector.go [hunk 1]
```diff
@@ -274,7 +281,16 @@ func (e *Elector) sendErr(err error) {
 
 // sendEvent sends the specified event on the events channel
 func (e *Elector) sendEvent(event Event) {
-	e.events <- event
+	select {
+	case e.events <- event:
+	case <-e.closing:
+		select {
```
Comment (gpaul) [inline elector/elector.go]: Why the inner select? If we've closed, there's no more delivery guarantee.

Comment (jdef) [reply elector/elector.go]: you're right, there is no more delivery guarantee. this is my attempt at best-effort delivery. if you think this is too complex then we can probably get rid of it.

**Pull request merged**